src/Integration/NewConnectedMode/NewBindingWorkflow.cs

```diff
@@ -0,0 +1,157 @@
+using SonarLint.VisualStudio.Integration.Progress;
+using SonarLint.VisualStudio.Integration.Resources;
+using SonarLint.VisualStudio.Progress.Controller;
+
+namespace SonarLint.VisualStudio.Integration.NewConnectedMode
+{
+    // New connected mode:
+    // Handles binding a solution for new connected mode i.e. writes the
+    // SQ config file to the solution level. No changes are made to 
+    // individual projects, or to the solution file itself.
+
+    /// <summary>
+    /// Workflow execution for the bind command for the new connected mode
+    /// </summary>
+    internal class NewBindingWorkflow : IBindingWorkflow
+    {
+        private readonly IHost host;
+        private readonly BindCommandArgs bindingArgs;
+        private readonly IConfigurationProvider writer;
+
+        public NewBindingWorkflow(IHost host, BindCommandArgs bindingArgs, IConfigurationProvider configWriter)
+        {
+            if (host == null)
+            {
+                throw new ArgumentNullException(nameof(host));
+            }
+
+            if (bindingArgs == null)
+            {
+                throw new ArgumentNullException(nameof(bindingArgs));
+            }
+
+            if (configWriter == null)
+            {
+                throw new ArgumentNullException(nameof(configWriter));
+            }
+
+            Debug.Assert(bindingArgs.ProjectKey != null);
+            Debug.Assert(bindingArgs.ProjectName != null);
+            Debug.Assert(bindingArgs.Connection != null);
+
+            this.host = host;
+            this.bindingArgs = bindingArgs;
+            this.writer = configWriter;
+        }
+
+        #region Workflow state
+
+        internal bool Succeeded { get; set; }
+
+        #endregion
+
+        #region Workflow startup
+
+        public IProgressEvents Run()
+        {
+            Debug.Assert(this.host.ActiveSection != null, "Expect the section to be attached at least until this method returns");
+
+            IProgressEvents progress = ProgressStepRunner.StartAsync(this.host,
+                this.host.ActiveSection.ProgressHost,
+                controller => this.CreateWorkflowSteps(controller));
+
+            this.DebugOnly_MonitorProgress(progress);
+
+            return progress;
+        }
+
+        [Conditional("DEBUG")]
+        private void DebugOnly_MonitorProgress(IProgressEvents progress)
+        {
+            progress.RunOnFinished(r => this.host.Logger.WriteLine("DEBUGONLY: New binding workflow finished, Execution result: {0}", r));
+        }
+
+        private ProgressStepDefinition[] CreateWorkflowSteps(IProgressController controller)
+        {
+            StepAttributes HiddenNonImpactingBackgroundStep = StepAttributes.BackgroundThread | StepAttributes.Hidden | StepAttributes.NoProgressImpact;
+
+            return new ProgressStepDefinition[]
+            {
+                //new ProgressStepDefinition(null, HiddenNonImpactingBackgroundStep,
```

Comment:
TODO: I'm experimenting with having a single step that does everything, since the the _begin_ and _end_ steps only write messages. Depending on what the user experience is like, I'll either delete these commented out lines, or go back to having a three separate steps.

Reply:
...in other respects, this class is very similar too but much simpler than the existing _BindingWorkflow_ class.